gpu/features/test/test_normals.cpp

```diff
@@ -371,6 +371,93 @@ TEST(PCL_FeaturesGPU, normals_highlevel_4)
     }
 }
 
+// Test from issue:
+// - https://github.com/PointCloudLibrary/pcl/issues/2371#issuecomment-577727912
+TEST(PCL_FeaturesGPU, issue_2371)
+{
+    // This number is magic, do not set to lower value.
+    // It may affect error reproducibility.
+    const std::size_t N = 1000;
+    std::vector<pcl::PointXYZ> cloud_cpu(N, {0.0, 0.0, 0.0});
+
+    pcl::gpu::NormalEstimation::PointCloud cloud_gpu;
+    cloud_gpu.upload(cloud_cpu);
+
+    pcl::gpu::NormalEstimation ne_gpu;
+    ne_gpu.setInputCloud(cloud_gpu);
+
+    const float radius_search = 2.0F;
+    const int max_results = 500;
+    ne_gpu.setRadiusSearch(radius_search, max_results);
+
+    pcl::gpu::NormalEstimation::Normals normals_gpu;
+    ne_gpu.compute(normals_gpu);
+}
+
+TEST(PCL_FeaturesGPU, normals_nan_gpu)
```

Comment:
Is it possible to break out the 2 additional tests `normals_nan{gpu,cpu}` in a different PR? That'll allow focus only on solving the failing test (IIRC, this test specifically will fail, right?)

Reply:
> this test specifically will fail, right?

Yes, `TEST(PCL_FeaturesGPU, normals_nan_gpu)` will fail when run. But `test_normals.cpp` is not used anywhere currently, so it will not affect CI or CTest suite.

Reply:
> But test_normals.cpp is not used anywhere currently

😢 I'll see what can be done to rectify this. Might take a few months though.